src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -40,33 +40,51 @@ public AsynchronousOperationListener(string featureName, bool enableDiagnosticTo
             TrackActiveTokens = Debugger.IsAttached || enableDiagnosticTokens;
         }
 
-        public async Task<bool> Delay(TimeSpan delay, CancellationToken cancellationToken)
+        public Task<bool> Delay(TimeSpan delay, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
Could you please add some comments on why we'd go though the trouble to implement it this way?

Reply:
Added `[PerformanceSensitive]` with a text constraint.